src/node.rs

```diff
@@ -245,7 +245,10 @@ mod test {
             let EventType::Decide { leaf_chain, .. } = event.event else {
                 continue;
             };
-            for (leaf, vid) in leaf_chain.iter().rev() {
```

Comment:
Seem to be a lot of cloning going on. I think you could de-structure here and remove some of those. 

```
for LeafInfo { leaf, vid, .. } in leaf_chain.iter().rev()
```
Looks like things are `Arc`ed so maybe it doesn't matter?

Reply:
Done. Yes, I thought the same, but for visibility I think this makes more sense.